### PR TITLE
Generate `plugin_firmware_index.json` with new boards data

### PR DESCRIPTION
**Pull request by MatteoPologruto**
Generate a new index for new boards named `plugin_firmware_index.json`, with the boards as the following entry:
```
  {
    "fqbn": "arduino:renesas_uno:unor4wifi",
    "firmware": [
      {
        "version": "0.1.0",
        "url": "https://downloads.arduino.cc/arduino-fwuploader/firmwares/ESP32-S3/0.1.0/ESP32-S3.bin",
        "checksum": "SHA-256:37c4f003373448504d49d5e3c1754d1d90b9bba3f8f3b529b435d331ca6900b1",
        "size": "3951616",
        "module": "ESP32-S3"
      },
      {
        "version": "0.2.0",
        "url": "https://downloads.arduino.cc/arduino-fwuploader/firmwares/ESP32-S3/0.2.0/ESP32-S3.bin",
        "checksum": "SHA-256:cb535826ba96d4f51bdac33d252c24bd9c692a7bef56710ec0431979cef09fc3",
        "size": "3951616",
        "module": "ESP32-S3"
      }
    ],
    "uploader_plugin": "arduino:uno-r4-wifi-fwuploader@1.0.0",
    "additional_tools": [
      "arduino:espflash@2.0.0",
      "arduino:bossac@1.9.1-arduino5"
    ],
    "module": "ESP32-S3",
    "name": "Arduino Uno R4 WiFi"
  }
```
The old `module_firmware_index.json` can still be generated using the `--no-new` flag when launching the script.

### CODE REVIEW

File: .github/workflows/generate-index.yml
Comment (umbynos) [inline .github/workflows/generate-index.yml:1]: Why is this required?

File: generator/generator.py [hunk 1]
```diff
@@ -284,7 +284,8 @@ def generate_boards_json(input_data, arduino_cli_path):
                 boards[fqbn]["name"] = board["name"]
                 break
 
-        boards[fqbn].update(create_upload_data(fqbn, installed_cores))
+        if fqbn not in skip_precomp:
```
Comment (umbynos) [inline generator/generator.py:297]: I would change the `skip_precom` variable name, since this is used for this too.

File: .github/workflows/generate-index.yml
Comment (MatteoPologruto) [reply .github/workflows/generate-index.yml:1]: For the older boards, the uploader data is generated through the `platform.txt` of the installed core, so [this check](https://github.com/arduino/arduino-fwuploader/blob/main/generator/generator.py#L253-L217) was added.
Now that I think about it, we could probably skip it for new boards and avoid installing unnecessary cores.

**Comment by umbynos** [comment]
I replaced the "placeholder firmware" with [0.1.0](https://github.com/bcmi-labs/uno-r4-wifi-usb-bridge/releases/tag/0.1.0) elegantly slapped together with [this script](https://github.com/bcmi-labs/uno-r4-wifi-usb-bridge/blob/main/combine.py) and [0.2.0](https://github.com/bcmi-labs/uno-r4-wifi-usb-bridge/releases/tag/0.2.0)

**Comment by umbynos** [comment]
Actually with @cmaglie we decided it was best to have two separate json files. The old `module_firmware_index.json` and a new `plugin_firmware_index.json` file: containing only the new plugin supported boards. This way we can avoid showing unsupported board in case of `arduino-fwuploader firmware list` with an old version of the fw uploader. The `generator.py` and the CI should consider this.

**Review by alessio-perugini — APPROVED** [review]

File: .github/workflows/generate-index.yml [hunk 2]
```diff
@@ -56,9 +56,12 @@ jobs:
           cd $GITHUB_WORKSPACE
           task poetry:install-deps
 
-      - name: Generate index
+      - name: Generate plugin firmware index
         run: poetry run ./generator.py -a $(which arduino-cli)
 
+      - name: Generate module firmware index
+        run: poetry run ./generator.py -a $(which arduino-cli) --no-new
```
Comment (umbynos) [inline .github/workflows/generate-index.yml:63]: wasn't the new flag introduced called `--new`?

File: .github/workflows/generate-index.yml [hunk 3]
```diff
@@ -69,16 +72,24 @@ jobs:
...
             --detach-sign boards/module_firmware_index.json
 
+      - name: sign the plugin firmware index json
+        run: |
+          gpg \
+            --pinentry-mode=loopback \
+            --passphrase "${{ secrets.PASSPHRASE }}" \
+            --output boards/plugin_firmware_index.json.sig \
+            --detach-sign boards/plugin_firmware_index.json
+
```
Comment (umbynos) [inline .github/workflows/generate-index.yml:90]: Isn't it possible to use a single command line?

**Review by umbynos — CHANGES_REQUESTED** [review]

File: .github/workflows/generate-index.yml (hunk 3 above)
Comment (umbynos) [reply .github/workflows/generate-index.yml:90]: I answered myself to that too :rofl: Since we are specifying `--output` we cannot have multiple `.sig` output files

**Review by umbynos — APPROVED** [review]

File: .github/workflows/generate-index.yml (hunk 2 above)
Comment (MatteoPologruto) [reply .github/workflows/generate-index.yml:63]: Using `--new` sets the flag to `true` (which is actually the default, so the behaviour with no flags is the same), while `--no-new` sets it to false. Should I change the names to make it clearer?

File: .github/workflows/generate-index.yml (hunk 3 above)
Comment (MatteoPologruto) [reply .github/workflows/generate-index.yml:90]: I think `--output` accepts only one argument, so I split the command line to avoid errors.
